dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/dataapproval/DataApprovalPermissionsEvaluator.java

```diff
@@ -261,6 +261,13 @@ public void evaluatePermissions( DataApprovalStatus status, DataApprovalWorkflow
         permissions.setMayAccept( mayAccept );
         permissions.setMayUnaccept( mayUnaccept );
         permissions.setMayReadData( mayReadData );
+        permissions.setApprovedAt( status.getCreated() );
+
+        if ( status.getState() == DataApprovalState.APPROVED_HERE
+            || (status.getState() == DataApprovalState.APPROVED_ABOVE && userLevelIndex < dataLevelIndex) )
+        {
+            permissions.setApprovedBy( status.getCreator() != null ? status.getCreator().getName() : "" );
```

Comment:
Perhaps cleaner to set null instead of empty string ("") if not set?

Reply:
agree, fixed now